crates/zizmor/src/fix.rs

```diff
@@ -0,0 +1,186 @@
+                        fix,
+                    ));
+                }
+            }
+        }
+    }
+
+    if file_fixes.is_empty() {
+        println!("No fixes available to apply.");
+        return Ok(());
+    }
+
+    // Process each file
+    let mut applied_fixes = Vec::new();
+    let mut failed_fixes = Vec::new();
+
+    for (file_path, fixes) in &file_fixes {
+        // Get the original content from the registry instead of reading from disk
+        let input_key = fixes
+            .first()
+            .unwrap()
+            .1
+            .locations
+            .first()
+            .unwrap()
+            .symbolic
+            .key;
+        let input = registry.get_input(input_key);
+        let original_content = input.as_document().source();
+
+        let mut current_content = original_content.to_string();
+        let mut file_applied_fixes = Vec::new();
+        let mut successful_fixes = Vec::new();
+
+        // First, try to apply each fix independently to the original content
+        // to collect which fixes can be applied successfully
+        for (ident, finding, fix) in fixes {
+            match fix.apply_to_content(&original_content) {
+                Ok(Some(_)) => {
+                    successful_fixes.push((*ident, *fix, *finding));
+                }
+                Ok(None) => {
+                    // Fix didn't apply (no changes needed)
+                }
+                Err(e) => {
+                    failed_fixes.push((*ident, file_path.clone(), format!("{}", e)));
+                }
+            }
+        }
+
+        // Then apply successful fixes sequentially, handling conflicts gracefully
+        for (ident, fix, finding) in successful_fixes {
+            match fix.apply_to_content(&current_content) {
+                Ok(Some(new_content)) => {
+                    current_content = new_content;
+                    file_applied_fixes.push((ident, fix, finding));
+                }
+                Ok(None) => {
+                    // Fix didn't apply to modified content (possibly due to conflicts)
+                }
+                Err(e) => {
+                    // If the fix fails on modified content, it might be due to conflicts
+                    // with previously applied fixes. Record this as a failed fix.
+                    failed_fixes.push((
+                        ident,
+                        file_path.clone(),
+                        format!("conflict after applying previous fixes: {}", e),
+                    ));
+                }
+            }
+        }
+
+        // Only proceed if there are changes to apply
+        if current_content != original_content {
+            println!("{}", format!("\nFixes").green().bold());
+            let num_fixes = file_applied_fixes.len();
+            for (ident, fix, finding) in file_applied_fixes {
+                let line_info = if let Some(line) = get_primary_line_number(finding) {
```

Comment:
FYI you can unwrap here, rather than handling a superfluous `None` case -- the finding builder APIs enforce the presence of a primary location, so it's safe to assume one here as well 🙂 

(Alternatively you could do the unwrap within `get_primary_line_number`, that might be a little cleaner.)